tests/dynamic_data/users_roles/delete_user_test.py

```diff
@@ -103,6 +114,90 @@ def test_delete_user_negative(setup_users_for_testing_errors):
         assert exc_info.value.http_code == expected_status
 
 
+@pytest.mark.usefixtures('ensure_layman_module', 'oauth2_provider_mock')
+def test_delete_own_public_publications():
+    # create publication only with owners rights
+    username = "test_delete_owner_only"
+    publication = 'test_delete_user_publication_owner'
+    public_workspace = 'test_workspace'
+
+    process_client.reserve_username(username, actor_name=username)
+    for publication_type in process_client.PUBLICATION_TYPES:
+        process_client.publish_workspace_publication(publication_type, public_workspace, publication, actor_name=username)
+
+    process_client.delete_user(username, actor_name=username)
+
+    for publication_type in process_client.PUBLICATION_TYPES:
+        with app.app_context():
+            publ_info = layman_util.get_publication_info(public_workspace, publication_type, publication)
+        assert not publ_info, f"Publication {publication} in workspace {public_workspace} was not deleted"
+
+
+@pytest.mark.usefixtures('ensure_layman_module', 'oauth2_provider_mock')
+def test_delete_shared_publications():
```

Comment:
We can parametrize with `@pytest.mark.parametrize('publication_type', process_client.PUBLICATION_TYPES)` here.